src/plugins/GUITestBase/src/harness/UGUITest.h

```diff
@@ -110,16 +112,54 @@ class UGUITest : public GUITest {
 #define SUITENAME(className) QString(GUI_TEST_SUITE)
 
 #define GUI_TEST_CLASS_DECLARATION(className) \
-    class className : public UGUITest { \
+    class className : public UGUITest \
+    { \
+        using UGUITest::UGUITest; \
+\
     public: \
-        className(int timeout = DEFAULT_GUI_TEST_TIMEOUT, const QStringList& labelList = QStringList()) \
-            : UGUITest(TESTNAME(className), SUITENAME(className), timeout, labelList.toSet()) { \
-        } \
+        className(int timeout = DEFAULT_GUI_TEST_TIMEOUT, \
+                  const QStringList &labelList = QStringList()) \
+            : UGUITest(TESTNAME(className), SUITENAME(className), timeout, labelList.toSet()) \
+        {} \
+\
+    protected: \
+        void run(HI::GUITestOpStatus &os) override; \
+    };
+
+#define GUI_TEST_CLASS_DECLARATION_MSA_MODE(className, mode) \
+    class className##_MSA_##mode : public className \
+    { \
+        using className::className; \
+\
+    public: \
+        className##_MSA_##mode(int timeout = DEFAULT_GUI_TEST_TIMEOUT, \
+                           const QStringList &labelList = QStringList()) \
+            : className(TESTNAME(className##_MSA_##mode), \
+                        SUITENAME(className##_MSA_##mode), \
+                        timeout, \
+                        labelList.toSet()) \
+        {} \
 \
     protected: \
-        void run(HI::GUITestOpStatus& os) override; \
+        void run(HI::GUITestOpStatus &os) override \
+        { \
+            Settings *s = AppContext::getSettings(); \
+            if (s != nullptr) { \
+                if (QString(#mode) == "MM") { \
+                    s->setValue(QString("msaeditor/") + "multiline_mode", true); \
+                } else { \
+                    s->setValue(QString("msaeditor/") + "multiline_mode", false); \
+                } \
+            } \
+            className::run(os); \
+        } \
     };
 
+#define GUI_TEST_CLASS_DECLARATION_MSA(className) \
+    GUI_TEST_CLASS_DECLARATION(className); \
+    GUI_TEST_CLASS_DECLARATION_MSA_MODE(className, SM); \
+    GUI_TEST_CLASS_DECLARATION_MSA_MODE(className, MM); \
```

Comment:
Please remove tests duplication: Multiline mode should have dedicated tests.
If we follow this pattern we have to duplicate all tests everywhere when new feature is added? Like annotation creation with DetView/PanView/Overview ON/OFF, ProjectView opened/closed, Options panel opened or closed.

Reply:
Can't agree now, maybe later
The single mode tests must be passed in the multi-line mode too
I have found ~30 bugs by this simple feature
Let's run for a while this duplicated tests
I will create additional explicit multi-line tests by the fails
After merge to a master we will switch the feature off

Reply:
You can keep these tests for your local development but please remove them before merging to the master

Reply:
done